src/gcp_airflow_foundations/source_class/ftp_gcs_source.py

```diff
@@ -0,0 +1,217 @@
+        destination_table = f"{gcp_project}:{landing_dataset}.{table_config.landing_zone_table_name_override}" + f"_{ds}"
+
+        schema_file_name = None
+        if "schema_file" in table_config.extra_options.get("gcs_table_config"):
+            schema_file_name = table_config.extra_options.get("gcs_table_config")["schema_file"]
+
+        # Get files to load from metadata file
+        if schema_file_name:
+            schema_file = self.gcs_hook.download(bucket_name=bucket, object_name=schema_file_name)
+
+            # Only supports json schema file format - add additional support if required
+            schema_fields = json.loads(schema_file)
+            logging.info(schema_fields)
+            gcs_to_bq = GCSToBigQueryOperator(
+                task_id='import_files_to_bq_landing',
+                bucket=bucket,
+                source_objects=files_to_load,
+                source_format=source_format,
+                schema_fields=schema_fields,
+                destination_project_dataset_table=destination_table,
+                write_disposition='WRITE_TRUNCATE',
+                create_disposition='CREATE_IF_NEEDED',
+                skip_leading_rows=1,
+            )
+        else:
+            gcs_to_bq = GCSToBigQueryOperator(
+                task_id='import_files_to_bq_landing',
+                bucket=bucket,
+                source_objects=files_to_load,
+                source_format=source_format,
+                destination_project_dataset_table=destination_table,
+                write_disposition='WRITE_TRUNCATE',
+                create_disposition='CREATE_IF_NEEDED',
+                skip_leading_rows=1,
+            )
+        gcs_to_bq.execute(context=kwargs)
+
+    def get_list_of_files(self, table_config, **kwargs):
+        # XCom push the list of files
+        bucket = self.config.source.extra_options["gcs_bucket"]
+        if "metadata_file" in table_config.extra_options.get("gcs_table_config"):
+            metadata_file_name = table_config.extra_options.get("gcs_table_config")["metadata_file"]
+            metadata_file = self.gcs_hook.download(bucket_name=bucket, object_name=metadata_file_name, filename="metadata.csv")
+            file_list = []
+            with open('metadata.csv', newline='') as f:
+                for line in f:
+                    file_list.append(line.strip())
+        else:
+            templated_file_name = self.config.source.extra_options["gcs_source_config"]["templated_file_name"]
+            templated_file_name = templated_file_name.replace("{{ TABLE_NAME }}", table_config.landing_zone_table_name_override)
+            file_list = [templated_file_name]
+
+        # support replacing files with current dates
+        ds = kwargs["ds"]
+        file_list[:] = [file.replace("{{ ds }}", ds) if "{{ ds }}" in file else file for file in file_list]
+
+        if self.config.source.extra_options["gcs_source_config"]["file_prefix_filtering"]:
+            logging.info("HELLO")
```

Comment:
Hello to you too

Reply:
LOL oops

Reply:
sometime airflow bugs out and you resort to this...